Lib/typing.py

```diff
@@ -1057,6 +1114,214 @@ def __dir__(self):
         return list(set(super().__dir__()
                 + [attr for attr in dir(self.__origin__) if not _is_dunder(attr)]))
 
+
+def _is_unpacked_tuple(x):
```

Comment:
n00b question: Are type annotations allowed within `typing.py`? Would be good to have parameter and return types.

Reply:
`typing.py`, `importlib` -- and, soon, `tomllib`! -- are just about the only places you _will_ find type annotations inside the stdlib :)

Reply:
Yes, I added a few in typing.py recently. I think they'd be helpful here too just as documentation, though it's probably not feasible to run a static type checker on typing.py itself.

Reply:
Ahh, I was wondering about this too - good call for actually asking, Pradeep :) I've gone through and added type annotations for the new functions we've added.